src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs

```diff
@@ -50,8 +56,8 @@ static void UpdateText(this IUITextInput textInput, InputView inputView, bool is
 			{
 				// Re-calculate the cursor offset position if the text was modified by a Converter.
 				// but if the text is being set by code, let's just move the cursor to the end.
-				var cursorOffset = newText.Length - oldText.Length;
-				var cursorPosition = isEditing ? textInput.GetCursorPosition(cursorOffset) : newText.Length;
+				var cursorOffset = newTextLength - oldText.Length;
+				var cursorPosition = isEditing ? textInput.GetCursorPosition(cursorOffset) : newTextLength;
```

Comment:
There are some cursor position tests failing on macOS:
![image](https://github.com/dotnet/maui/assets/6755973/01af0af2-6f25-46cd-b1da-9d906014a2bb)

Reply:
Example: 
![image](https://github.com/dotnet/maui/assets/6755973/c8e9802f-e803-427c-a0cd-42f37c317427)

Reply:
I added a commit that fixes that